routes/payment_intent/index.mjs

```diff
@@ -0,0 +1,297 @@
+  const {
+    from_number,
+    to_number,
+    amount,
+    currency,
+    description,
+    cancelation_reason,
+    payment_method,
+    amount_received,
+  } = await c.req.json().catch(() => ({}));
+  const secret = `client_secret_${from_number}`;
+
+  try {
+    const { data, error } = await supabase
+      .from("intents")
+      .update({
+        from_number: from_number,
+        to_number: to_number,
+        amount: amount,
+        currency: currency,
+        description: description,
+        cancelation_reason: cancelation_reason,
+        payment_method: payment_method,
+        amount_received: amount_received,
+      })
+      .eq("id", id)
+      .eq("client_secret", secret)
+      .select();
+    if (error) {
+      console.log(error);
+    }
+
+    return c.json(data, 201);
+  } catch (err) {
+    console.error("Unexpected error:", err);
+    return c.json(
+      {
+        message: "Unexpected error occurred",
+        error: err.message,
+      },
+      500
+    );
+  }
+});
+
+//get payment intents ----- This is not clear are we getting all the client's intent or all the intents in the DB
+app.get("/", async (c) => {
+  try {
+    const { data, error } = await supabase.from("intents").select();
+    if (error) {
+      console.log(error);
+    }
+
+    return c.json(data, 201);
+  } catch (err) {
+    console.error("Unexpected error:", err);
+    return c.json(
+      {
+        message: "Unexpected error occurred",
+        error: err.message,
+      },
+      500
+    );
+  }
+});
+
+//get payment intent by id
+app.get("/:id", async (c) => {
+  const id = c.req.param("id");
+
+  if (!id) {
+    return c.json(
+      {
+        message: "Invalid request; missing `id`",
+      },
+      400
+    );
+  }
+
+  try {
+    const { data, error } = await supabase
+      .from("intents")
+      .select()
+      .eq("id", id);
+    if (error) {
+      console.log(error);
+    }
+
+    return c.json(data, 201);
+  } catch (err) {
+    console.error("Unexpected error:", err);
+    return c.json(
+      {
+        message: "Unexpected error occurred",
+        error: err.message,
+      },
+      500
+    );
+  }
+});
+
+//cancel payment intent
+app.put("/:id/cancel", async (c) => {
+  const id = c.req.param("id");
+
+  if (!id) {
+    return c.json(
+      {
+        message: "Invalid request; missing `id`",
+      },
+      400
+    );
+  }
+
+  try {
+    const { data, error } = await supabase
+      .from("intents")
+      .update({
+        cancelled_at: new Date(),
+        cancellation_reason: "User cancelled",
+      })
+      .eq("id", id);
+    if (error) {
+      console.log(error);
+    }
+
+    return c.json("Intent cancelled!", 201);
+  } catch (err) {
+    console.error("Unexpected error:", err);
+    return c.json(
+      {
+        message: "Unexpected error occurred",
+        error: err.message,
+      },
+      500
+    );
+  }
+});
+
+//confirm payment intent
+app.put("/:id/confirm", async (c) => {
```

Comment:
there may need to be additional checks when confirming an intent.

e.g. do we have all the required fields